src/Microsoft.AspNetCore.Antiforgery/Internal/DefaultAntiforgeryTokenSerializer.cs

```diff
@@ -156,7 +161,14 @@ public string Serialize(AntiforgeryToken token)
 
                 writer.Flush();
                 var stream = serializationContext.Stream;
-                return WebEncoders.Base64UrlEncode(_cryptoSystem.Protect(stream.ToArray()));
+                var bytes = _cryptoSystem.Protect(stream.ToArray());
```

Comment:
BTW this allocation alone is not huge, likely because it occurs prior to encryption and URL-encoding, both of which expand. Just a bit more than a quarter the `byte`s allocated in this `Protect()` call for example:
![after bytes](https://cloud.githubusercontent.com/assets/6431421/13099084/a336933e-d4e3-11e5-9fa6-984d8b16c1b9.png)